PharoProjects/Announcer.md

```diff
@@ -0,0 +1,112 @@
+# Announcer
+`Announcer` and `Announcement` classes provide an implementation of the [observer design pattern](https://en.wikipedia.org/wiki/Observer_pattern) in Pharo.
+As written on the wikipedia page, the observer design pattern (and thus `Announcers` and `Announcements`) aims to address the following problems:
+
+> - A one-to-many dependency between objects should be defined without making the objects tightly coupled.
+> - It should be ensured that when one object changes state an open-ended number of dependent objects are updated automatically.
+> - It should be possible that one object can notify an open-ended number of other objects.
+
+This page provide a quick tutorial to get started with them.
+
+- [Getting started](#getting-started)
+- [Defining new kind of Announcement](#defining-new-kind-of-announcement)
+- [Using announcer for your subject object](#using-announcer-for-your-subject-object)
+
+## Getting started
+The announcer class implements a mechanism to:
+1. Allow objects (the observers) to subscribe to announcements;
+2. Manage objects subscriptions; and
+3. Allow the object holding the announcer (the subject) to announce events.
+
+For example, there is an announcer available to let objects listen to announcements (events) concerning the system: `SystemAnnouncer uniqueInstance`.
+Such announcements include:
+- `ClassAdded`
+- `ClassRemoved`
+- `MethodModified`
+- etc.
+
+If one wants to be notified when a class is removed, it is possible to subscribe an object to this specific announcement as follow:
+
+First, let's create the method `#whenClassAdded:` for the object that will listen to `ClassAdded` announcement.
+This method will be the hook called when such announcement has been announced.
+It takes the announcement sent by the subject as parameter.
+
+```Smalltalk
+MyObjectListeningToClassAdded>>#whenClassAdded: aClassAdded
+  Transcript
+    show: aClassAdded classAffected name;
+    show: ' has been added.';
+    cr
+```
+
+Then, we subscribe an instance of our object to the announcer.
+When doing that, it is needed to specify
+1. Which kind of announcements (its class) our object listen to.
+2. Which object list to this kind of announcements.
+3. Which method to call when the annoncement is sent.
+
+The previous step can be achieved as follow:
+
+```Smalltalk
+SystemAnnouncer uniqueInstance
+  when: ClassAdded send: #whenClassAdded: to: instanceOfMyObjectListeningToClassAdded
```

Comment:
We could also talk about #when:do:

Reply:
Yes but it should be done in the future because we should discuss that when giving a `Block` it create an "anonymous" observer and it is difficult to unsubscribe it. :-)